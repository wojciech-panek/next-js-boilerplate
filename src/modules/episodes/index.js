export {
  EpisodesTypes,
  EpisodesActions,
} from './episodes.redux';

export * from './episodes.selectors';
