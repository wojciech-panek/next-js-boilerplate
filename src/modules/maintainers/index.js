export {
  MaintainersTypes,
  MaintainersActions,
} from './maintainers.redux';

export * from './maintainers.selectors';
