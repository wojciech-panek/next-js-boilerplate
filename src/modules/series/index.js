export {
  SeriesTypes,
  SeriesActions,
} from './series.redux';

export * from './series.selectors';
