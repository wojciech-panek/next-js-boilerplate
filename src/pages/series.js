import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'next/router';
import { compose } from 'ramda';

import withIntl from '../shared/lib/withIntl';


export class SeriesPage extends PureComponent {
  static propTypes = {
    router: PropTypes.object.isRequired,
  };

  render() {
    return (
      <div>Series: {this.props.router.query.id}</div>
    );
  }
}

export default compose(
  withIntl,
  withRouter,
)(SeriesPage);
