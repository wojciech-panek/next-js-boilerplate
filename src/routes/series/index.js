export { default as Series } from './series.container';
