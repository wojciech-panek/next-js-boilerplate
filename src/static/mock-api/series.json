[
  {
    "id": 1,
    "title": "Dick in Ibiza",
    "season": "1"
  },
  {
    "id": 2,
    "title": "Dick in Ibiza",
    "season": "2"
  },
  {
    "id": 3,
    "title": "Liars",
    "season": "1"
  },
  {
    "id": 4,
    "title": "F*ck My Life",
    "season": "1"
  }
]
