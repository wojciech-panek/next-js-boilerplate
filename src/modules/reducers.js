import { combineReducers } from 'redux-immutable';

import { reducer as localesReducer } from './locales/locales.redux';
import { reducer as seriesReducer } from './series/series.redux';
import { reducer as episodesReducer } from './episodes/episodes.redux';
//<-- IMPORT MODULE REDUCER -->

export default function createReducer() {
  return combineReducers({
    series: seriesReducer,
    episodes: episodesReducer,
    locales: localesReducer,
    //<-- INJECT MODULE REDUCER -->
  });
}
