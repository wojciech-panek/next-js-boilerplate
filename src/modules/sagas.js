import { all, fork } from 'redux-saga/effects';

import reportError from '../shared/utils/reportError';
import { watchSeries } from './series/series.sagas';
import { watchEpisodes } from './episodes/episodes.sagas';
//<-- IMPORT MODULE SAGA -->


export default function* rootSaga() {
  try {
    yield all([
      fork(watchSeries),
      fork(watchEpisodes),
      //<-- INJECT MODULE SAGA -->
    ]);
  } catch (e) {
    yield reportError(e);
  }
}
