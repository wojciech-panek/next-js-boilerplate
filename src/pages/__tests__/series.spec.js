import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import { expect } from 'chai';

import { SeriesPage } from '../series';
import { SeriesActions } from '../../modules/series';


describe('Series: Page', () => {
  const defaultProps = {
    router: { query: { id: 'test-id' } },
  };

  const component = (props) => (
    <SeriesPage {...defaultProps} {...props} />
  );

  it('should render correctly', () => {
    const wrapper = shallow(component());
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch SeriesActions.fetch in getInitialProps on server', async () => {
    const dispatch = spy();
    const query = { id: '1' };
    const store = global.utils.prepareInitialPropsStore(dispatch);

    await SeriesPage.getInitialProps({ isServer: true, store, query });

    expect(dispatch).to.have.been.calledOnce;
    expect(dispatch).to.have.been.calledWith(SeriesActions.fetchSingle(1));
  });

  it('should dispatch SeriesActions.fetch in getInitialProps on client', async () => {
    const query = { id: '1' };
    const store = {
      dispatch: spy(),
    };

    await SeriesPage.getInitialProps({ isServer: false, store, query });

    expect(store.dispatch).to.have.been.calledOnce;
    expect(store.dispatch).to.have.been.calledWith(SeriesActions.fetchSingle(1));
  });
});
