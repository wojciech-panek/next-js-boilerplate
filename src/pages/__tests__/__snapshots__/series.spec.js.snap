// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Series: Page should render correctly 1`] = `<InjectIntl(withRouter(Connect(Series))) />`;
